src/config/validation.rs

```diff
@@ -7,10 +7,27 @@ pub fn check_shape(config: &Config) -> Result<(), Vec<String>> {
     if config.sources.is_empty() {
         errors.push("No sources defined in the config.".to_owned());
     }
+
     if config.sinks.is_empty() {
         errors.push("No sinks defined in the config.".to_owned());
     }
 
+    // Check for non-unique names across sources, sinks, and transforms
+    let mut name_counts = HashMap::<&str, usize>::new();
```

Comment:
Would it be worth trying to tell the user which kind of component was colliding? Then we'd collect into a `HashMap::<&str, Vec<ComponentKind>>`.